Penumbra/Import/Textures/CombinedTexture.Manipulation.cs

```diff
@@ -206,26 +243,59 @@ private void MultiplyPixelsRight(int y, ParallelLoopState _)
         }
     }
 
+    private byte[] ResizePixels(byte[] rgbaPixels, int sourceWidth, int sourceHeight)
+    {
+        if (sourceWidth == _targetWidth && sourceHeight == _targetHeight)
+            return rgbaPixels;
+
+        byte[] resizedPixels;
+        using (var image = Image.LoadPixelData<Rgba32>(rgbaPixels, sourceWidth, sourceHeight))
+        {
+            image.Mutate(ctx => ctx.Resize(_targetWidth, _targetHeight));
+
+            resizedPixels = new byte[_targetWidth * _targetHeight * 4];
+            image.CopyPixelDataTo(resizedPixels);
+        }
+
+        return resizedPixels;
+    }
+
 
     private (int Width, int Height) CombineImage()
     {
         var combineOp = GetActualCombineOp();
-        var (width, height) = combineOp is not CombineOp.Invalid or CombineOp.RightCopy or CombineOp.RightMultiply
+        var (usesLeft, usesRight) = GetCombineOpFlags(combineOp);
+        var resizeOp = usesLeft && usesRight ? _resizeOp : ResizeOp.None;
+        (_targetWidth, _targetHeight) = usesLeft && resizeOp != ResizeOp.ToRight
```

Comment:
I'd prefer this code clearer. Maybe like this?
```
(_targetWidth, _targetHeight) = resizeOp switch
{
    ResizeOp.ToLeft =>  (_left.TextureWrap!.Width, _left.TextureWrap!.Height),
    ResizeOp.None when usesLeft => (_right.TextureWrap!.Width, _right.TextureWrap!.Height),
    ResizeOp.ToRight => (_right.TextureWrap!.Width, _right.TextureWrap!.Height),
    _ => (_right.TextureWrap!.Width, _right.TextureWrap!.Height),
}
```

Reply:
`None when usesLeft` shall use left's size, I guess it did need to be clearer :smile: